go/v1/resource_descriptor.go

```diff
@@ -5,29 +5,51 @@ Wrapper APIs for in-toto attestation ResourceDescriptor protos.
 package v1
 
 import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha512"
 	"encoding/hex"
 	"errors"
 )
 
 var (
+	ErrIncorrectDigestLength = errors.New("digest is not correct length")
```

Comment:
Since attestations can have lots of digests in them would it make sense to put more metadata in the errors that get returned?  Ideally we could return the digest type and value?  That way folks could more easily find the part of the attestation that's generating the error?

(This probably applies to both `ErrIncorrectDigestLength` and `ErrInvalidDigestEncoding`)

Reply:
We could definitely do this for the length error. `ErrInvalidDigestEncoding` is currently only returned if any of the standard supported algorithms isn't hex encoded, so not sure what other metadata we could add here other than have a more detailed error message?

Reply:
Oh, I just realized, you probably mean to add specific algorithm information to `ErrInvalidDigestEncoding`. Yes, that makes sense too!

Reply:
I ended up leaving these errors intact and instead parameterized the error messages in the `Validate()` function, as parametrizing these errors broke error checking in the tests.